common/src/main/java/net/adoptopenjdk/icedteaweb/jvm/JVMUtils.java

```diff
@@ -0,0 +1,113 @@
+package net.adoptopenjdk.icedteaweb.jvm;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class JVMUtils {
+    /**
+     * Check that the VM args are valid and safe.
+     *
+     * Based on
+     * http://java.sun.com/javase/6/docs/technotes/guides/javaws/developersguide/syntax.html
+     *
+     * @param vmArgs a string containing the args
+     * @throws IllegalArgumentException if the VM arguments are invalid or dangerous
+     */
+    public static void checkVMArgs(final String vmArgs) throws IllegalArgumentException {
+        if (vmArgs == null) {
+            return;
+        }
+
+        final List<String> validArguments = Arrays.asList(getValidVMArguments());
+        final String[] validStartingArguments = getValidStartingVMArguments();
+
+        final String[] arguments = vmArgs.split(" ");
+        boolean argumentIsValid;
+        for (String argument : arguments) {
+            argumentIsValid = false;
+
+            if (validArguments.contains(argument)) {
+                argumentIsValid = true;
+            } else {
+                for (String validStartingArgument : validStartingArguments) {
+                    if (argument.startsWith(validStartingArgument)) {
+                        argumentIsValid = true;
+                        break;
+                    }
+                }
+            }
+
+            if (!argumentIsValid) {
+                throw new IllegalArgumentException(argument);
+            }
+        }
+    }
+
+    /**
+     * Returns an array of valid (ie safe and supported) arguments for the JVM
+     *
+     * Based on
+     * http://java.sun.com/javase/6/docs/technotes/guides/javaws/developersguide/syntax.html
+     */
+    private static String[] getValidVMArguments() {
```

Comment:
could be converted to a constant.

Best would be `private static final Set<String> VALID_VM_ARGUEMNTS'

Reply:
according to zoom session: we do that in a separate PR